src/test/java/com/cloudant/tests/util/Utils.java

```diff
@@ -33,4 +44,61 @@ public static String getHostName(String account) {
         }
 
     }
+
+    public static void removeReplicatorTestDoc(CloudantClient account, Response replicatorResponse)
+            throws Exception {
+
+        //Find replicator document for removal
+        ReplicatorDocument replicatorDoc = waitForReplicatorToComplete(account, replicatorResponse);
+
+        Response removeResponse = account.replicator()
+                .replicatorDocId(replicatorDoc.getId())
+                .replicatorDocRev(replicatorDoc.getRevision())
+                .remove();
+
+        assertThat(removeResponse.getError(), is(nullValue()));
+    }
+
+    public static ReplicatorDocument waitForReplicatorToComplete(CloudantClient account, Response replicatorResponse)
+            throws Exception {
+        ReplicatorDocument replicatorDoc = null;
+
+        boolean finished = false;
+
+        while(!finished && !hasTimedOut()) {
+            //Sleep before finding replication document
+            Thread.sleep(TIMEOUT_CONSTANT);
+
+            replicatorDoc = account.replicator()
+                    .replicatorDocId(replicatorResponse.getId())
+                    .find();
+
+            //If replicator state is completed or triggered, return true
+            //If false, continue in the while loop until replicator is complete
+            if(replicatorDoc != null && replicatorDoc.getReplicationState() != null
+                    && (replicatorDoc.getReplicationState().equalsIgnoreCase("completed")
+                    || (replicatorDoc.getReplicationState().equalsIgnoreCase("triggered")
+                    && replicatorDoc.getContinuous()))) {
+
+                finished = true;
+            }
+        }
+
+        //Reset timeout to default 50 milliseconds
+        TIMEOUT_CONSTANT = 100;
+
+        return replicatorDoc;
+    }
+
+    private static boolean hasTimedOut() {
+        long currentTime = System.currentTimeMillis();
+        if(TIMEOUT_CONSTANT <= 3200 && currentTime < (currentTime + TIMEOUT_CONSTANT)) {
```

Comment:
The check `currentTime < (currentTime + TIMEOUT_CONSTANT))` is always going to be true if `TIMEOUT_CONSTANT` is >0. It is also a bit confusing for me that the `TIMEOUT_CONSTANT` is not a constant. How would you feel about removing this `hasTimedOut` method and changing the while loop in `waitForReplicatorToComplete` to something like:
```java
        long startTime = System.currentTimeMillis();
        long timeout = startTime + TIMEOUT_CONSTANT;
        //initial wait of 100 ms
        long delay = 100;
        while (!finished && System.currentTimeMillis() < timeout) {
            //Sleep before finding replication document
            Thread.sleep(delay);

            replicatorDoc = account.replicator()
                    .replicatorDocId(replicatorResponse.getId())
                    .find();

            //If replicator state is completed or triggered, return true
            //If false, continue in the while loop until replicator is complete

            if (replicatorDoc != null && replicatorDoc.getReplicationState() != null
                    && (replicatorDoc.getReplicationState().equalsIgnoreCase("completed")
                    || (replicatorDoc.getReplicationState().equalsIgnoreCase("triggered")
                    && replicatorDoc.getContinuous()))) {
                finished = true;
            }
            //double the delay for the next iteration
            delay *= 2;
        }
```